tests/unit/test_serialization.py

```diff
@@ -45,6 +48,121 @@ async def test_happy_flow(default_config_path, mock_registry_client):
         assert 1 in o.ids_to_schemas
 
 
+def test_avro_json_flatten_unions():
+    typed_schema = TypedSchema.parse(
+        SchemaType.AVRO,
+        json.dumps(
+            {
+                "namespace": "io.aiven.data",
+                "name": "Test",
+                "type": "record",
+                "fields": [
+                    {
+                        "name": "attr1",
+                        "type": ["null", "string"],
+                    },
+                    {
+                        "name": "attr2",
+                        "type": ["null", "string"],
+                    },
+                ],
+            }
+        ),
+    )
+    record = {"attr1": {"string": "sample data"}, "attr2": None}
+    flatten_record = {"attr1": "sample data", "attr2": None}
+    assert flatten_unions(typed_schema.schema, record) == flatten_record
+
+    record = {"attr1": None, "attr2": None}
+    assert flatten_unions(typed_schema.schema, record) == record
+
+    typed_schema = TypedSchema.parse(
+        SchemaType.AVRO,
+        json.dumps(
+            {
+                "type": "array",
+                "items": {
+                    "namespace": "io.aiven.data",
+                    "name": "Test",
+                    "type": "record",
+                    "fields": [
+                        {
+                            "name": "attr",
+                            "type": ["null", "string"],
+                        }
+                    ],
+                },
+            }
+        ),
+    )
+    record = [{"attr": {"string": "sample data"}}]
+    flatten_record = [{"attr": "sample data"}]
+    assert flatten_unions(typed_schema.schema, record) == flatten_record
```

Comment:
The None case could be tested too.

Reply:
what do you mean?

Reply:
ping

Reply:
ping

Reply:
Probably the case where `attr=None`.

Reply:
That is tested two lines down

Reply:
This hasn't receive a reply, marking as done.